ui-manchette-with-spacetimechart/src/hooks/usePaths.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from 'react';
+
+import {
+  type ProjectPathTrainResult,
+  type SpaceTimeCurves,
+} from '@osrd-project/ui-manchette/dist/types'; 
+
+const transformCurve = (curve: SpaceTimeCurves, departureTime: string) =>
+  curve.positions.map((position, i) => ({
+    time: curve.times[i] + new Date(departureTime).getTime(),
+    position,
+  }));
+
+const usePaths = (projectPathTrainResult: ProjectPathTrainResult[], selectedProjection?: number) =>
```

Comment:
It's tech debt, but can you rename selectedProjection by selectedTrain ? We don't want to highlight the train used for the projection, but the one selected
```suggestion
const usePaths = (projectPathTrainResult: ProjectPathTrainResult[], selectedTrain?: number) =>
```

Reply:
done